### PR TITLE
Fixing #218 WindowsError on non-windows systems

### LINKED ISSUE #218: Error message when a temporary file failed to be removed on Ubuntu, related to #150
# Description

I'have an error that should be a consequence of some modifications made for #150 :
```
Traceback (most recent call last):
  File "/home/afsy/jrs/Documents/geometwps/srvgeomet27/src/pywps-master/pywps/app/Process.py", line 227, in _run_process
    100, STATUS.DONE_STATUS, clean=self.async)
  File "/home/afsy/jrs/Documents/geometwps/srvgeomet27/src/pywps-master/pywps/app/WPSResponse.py", line 70, in update_status
    self.write_response_doc(self.doc, clean)
  File "/home/afsy/jrs/Documents/geometwps/srvgeomet27/src/pywps-master/pywps/app/WPSResponse.py", line 84, in write_response_doc
    self.process.clean()
  File "/home/afsy/jrs/Documents/geometwps/srvgeomet27/src/pywps-master/pywps/app/Process.py", line 281, in clean
    except WindowsError as err:
NameError: global name 'WindowsError' is not defined
```

# Environment

- operating system: Ubuntu 14.04, 16.04
- Python version: 2.7
- PyWPS version: 4, latest
- source/distribution
 - [x] git clone
 - [ ] Debian
 - [ ] PyPI
 - [ ] zip/tar.gz
 - [ ] other (please specify): 
- web server
 - [ ] Apache/mod_wsgi
 - [ ] CGI
 - [x] other (please specify): Flask

# Steps to Reproduce
Lock on the tmp dir
# Additional Information

### PR DESCRIPTION
**Pull request by jachym**
# Overview

Fixes #218 

# Related Issue / Discussion

Not so stable and robust database logging

# Additional Information

Changes a bit behaviour of database session management too

# Contribution Agreement

(as per https://github.com/geopython/pywps/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [x] I have already previously agreed to the PyWPS Contributions and Licensing Guidelines

### CODE REVIEW

File: pywps/exceptions.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class NoApplicableCode(HTTPException):
     Base exception class
     """
 
-    code = 400
+    code = 500
```
Comment (tomkralidis) [inline pywps/exceptions.py]: The default `code` param below is 400, yet the above line is changed to 500.  Do these need to be consistent?

File: pywps/app/Process.py [hunk 2]
```diff
@@ -278,10 +278,8 @@ def clean(self):
             if self._grass_mapset and os.path.isdir(self._grass_mapset):
                 LOGGER.info("Removing temporary GRASS GIS mapset: %s" % self._grass_mapset)
                 shutil.rmtree(self._grass_mapset)
-        except WindowsError as err:
-                LOGGER.error('Windows Error: %s', err)
         except Exception as err:
-                LOGGER.error('Unable to remove directory: %s', err)
+            LOGGER.error('Unable to remove directory: %s', err)
```
Comment (tomkralidis) [inline pywps/app/Process.py]: By deleting the `except` logic won't this result in a fatal error at runtime?

File: pywps/exceptions.py (hunk 1 above)
Comment (jachym) [inline pywps/exceptions.py]: 400 -> NotFound
500 -> ServerError 

which on this place is IMHO more appropriate (no applicable exception)

File: pywps/exceptions.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ class NoApplicableCode(HTTPException):
...
     locator = ""
 
     def __init__(self, description, locator="", code=400):
```
Comment (tomkralidis) [inline pywps/exceptions.py]: So here should `code=500`?

File: pywps/app/Process.py (hunk 2 above)
Comment (jachym) [inline pywps/app/Process.py]: `WindowsError` should be subclass of `Exception`, therefore the logic should remain the same. WindowsError is available on windows only - that caused some errors on other platforms (see #218)

this is actually core of the patch. other lines are just minor enhancements

File: pywps/exceptions.py (hunk 3 above)
Comment (jachym) [inline pywps/exceptions.py]: hmm makes sense

btw, just refer to https://en.wikipedia.org/wiki/List_of_HTTP_status_codes we should use them more

**Pull request merged**